src/main/kotlin/com/example/realworldkotlinspringbootjdbc/presentation/ArticleController.kt

```diff
@@ -242,97 +234,66 @@ class ArticleController(
-                }
-                /**
-                 * 記事取得 成功
-                 */
-                is Right -> {
-                    ResponseEntity(
-                        Article.from(showArticleResult.value).serializeWithRootName(),
-                        HttpStatus.valueOf(200),
-                    )
-                }
+    override fun getArticle(slug: String, authorization: String?): ResponseEntity<SingleArticleResponse> {
+        /**
+         * authorizationが
+         *   - 無い(null) -> None
+         *   - 有る -> 認証 -> 成功 -> Some<RegisteredUser>
+         *   - 有る -> 認証 -> 失敗 -> None
+         */
+        val currentUser = authorization.toOption().fold(
+            { none() },
+            {
+                realworldAuthenticationUseCase.execute(it).fold(
+                    { none() },
+                    { user -> Some(user) }
+                )
             }
+        )
 
-            /**
-             * JWT 認証 成功
-             */
-            is Right -> when (val showArticleResult = showArticle.execute(slug, Some(authorizeResult.value))) {
-                /**
-                 * 記事取得 失敗
-                 */
-                is Left -> when (showArticleResult.value) {
-                    /**
-                     * 原因: slug に該当する記事が見つからなかった
-                     */
-                    is ShowArticleUseCase.Error.NotFoundArticleBySlug -> ResponseEntity(
-                        serializeUnexpectedErrorForResponseBody("記事が見つかりませんでした"), // TODO: serializeUnexpectedErrorForResponseBodyをやめる
-                        HttpStatus.valueOf(404)
-                    )
-                    /**
-                     * 原因: バリデーションエラー
-                     */
-                    is ShowArticleUseCase.Error.ValidationErrors -> ResponseEntity(
-                        serializeUnexpectedErrorForResponseBody("記事が見つかりませんでした"), // TODO: serializeUnexpectedErrorForResponseBodyをやめる
-                        HttpStatus.valueOf(404)
-                    )
-                    /**
-                     * 原因: ユーザーがいなかった
-                     */
-                    is ShowArticleUseCase.Error.NotFoundUser -> ResponseEntity(
-                        serializeUnexpectedErrorForResponseBody("ユーザー登録されていませんでした"), // TODO: serializeUnexpectedErrorForResponseBodyをやめる
-                        HttpStatus.valueOf(404)
-                    )
-                }
-                /**
-                 * 記事取得 成功
-                 */
-                is Right -> {
-                    ResponseEntity(
-                        Article.from(showArticleResult.value).serializeWithRootName(),
-                        HttpStatus.valueOf(200),
+        val createdArticleWithAuthor = showArticle.execute(slug, currentUser)
+            .getOrHandle { throw ShowArticleUseCaseErrorException(it) }
+
+        return ResponseEntity(
+            SingleArticleResponse(
+                article = com.example.realworldkotlinspringbootjdbc.openapi.generated.model.Article(
```

Comment:
```suggestion
                article = Article(
```

フルパスでの指定が不要になったので削除をお願いします。合計 6 行あります。

Reply:
https://github.com/sunabak0/realworld-kotlin-springboot-jdbc/pull/323/commits/1d2613a1481abb876d39a18a94e0a9747876cc32

で対応しました

Reply:
確認しました。